packages/controllers/src/snaps/SnapController.ts

```diff
@@ -2156,11 +2158,24 @@ export class SnapController extends BaseController<
       const timer = new Timer(this._maxRequestTime);
       this._recordSnapRpcRequestStart(snapId, request.id, timer);
 
+      const handleRpcRequestPromise = this.messagingSystem.call(
+        'ExecutionService:handleRpcRequest',
+        snapId,
+        origin,
+        _request,
+      );
+
+      if (!handleRpcRequestPromise) {
```

Comment:
Is this possible? We should throw in `ExecutionService:handleRpcRequest` before this right?

Reply:
Yes that's right, this check was already in the handleRpcRequest in AbstractExecutionService, so it will be handled from there. I removed this check from the Controller.